src/coreclr/jit/lsraxarch.cpp

```diff
@@ -932,6 +932,16 @@ int LinearScan::BuildShiftRotate(GenTree* tree)
     {
         assert(shiftBy->OperIsConst());
     }
+#if defined(TARGET_64BIT)
```

Comment:
Is this instruction only applicable for x64? I thought it is also valid for x86? @tannergooding ?

Reply:
I wonder why it worked for x86 without changing this part of the code?  

> Is this instruction only applicable for x64? I thought it is also valid for x86? @tannergooding ?

Reply:
Isn't `InstructionSet_BMI2` the 32 bit version and for 64 bit you'd want `InstructionSet_BMI2_X64` ? If so and you checked 64 bit correctly elsewhere and then 32 here it would explain it working for 32.

Reply:
Double checked that my change handles only x64 case. Will open an issue to handle x86.